src/main/java/ch/wisv/areafiftylan/products/service/TicketServiceImpl.java

```diff
@@ -198,6 +202,15 @@ public void cancelTicketTransfer(String token) {
         return ticketRepository.findByPickupService_True();
     }
 
+    @Override
+    public Ticket assignTicketToUser(Long ticketId, String username) {
+        Ticket ticket = getTicketById(ticketId);
+        User user = userService.getUserByUsername(username);
+        ticket.setOwner(user);
+        return ticketRepository.save(ticket);
```

Comment:
Let's create a separate issue for this. It's probably worth investigating, but covers more then just this PR.